mne/viz/ica.py

```diff
@@ -38,6 +38,35 @@ def _ica_plot_sources_onpick_(event, sources=None, ylims=None):
         raise err
 
 
+def _butterfly_on_button_press(event, params):
+    """Helper to only draw once for picking"""
+    if params['need_draw']:
+        event.canvas.draw()
+    else:
+        text = params['texts'][0]
+        text.set_alpha(0.)
+        text.set_path_effects([])
+        event.canvas.draw()
+    params['need_draw'] = False
+
+
+def _butterfly_onpick(event, params):
+    """Helper to add a channel name on click"""
+    params['need_draw'] = True
+    ax = event.artist.get_axes()
+    lidx = np.where([l is event.artist for l in params['lines']])[0]
+    ch_name = params['ch_names'][lidx]
+    text = params['texts'][0]
+    x = event.artist.get_xdata()[event.ind[0]]
+    y = event.artist.get_ydata()[event.ind[0]]
+    text.set_x(x)
+    text.set_y(y)
+    text.set_text(ch_name)
+    text.set_color(event.artist.get_color())
+    text.set_alpha(1.)
+    text.set_path_effects(params['path_effects'])
```

Comment:
kinda, they have been slightly modified since there aren't subplots. I could make it so that it is `plt.subplot(111)` so I could just import these functions to remove the overlap